client/landing/domains/registrant-verification/index.jsx

```diff
@@ -0,0 +1,249 @@
+		};
+	};
+
+	getExpiredState = () => {
+		const { translate } = this.props;
+
+		return {
+			message: translate(
+				'This email has expired.{{br/}}Please resend the verification email and try again.',
+				{
+					components: {
+						br: <br />,
+					},
+				}
+			),
+			actionTitle: translate( 'Resend Email' ),
+			actionCallback: this.resendEmail,
+		};
+	};
+
+	getEmailMismatchState = () => {
+		const { domain, email, translate } = this.props;
+
+		return {
+			message: translate(
+				'{{strong}}%(email)s{{/strong}} is different from the email we have on record for {{strong}}%(domain)s{{/strong}}.',
+				{
+					args: {
+						email: email,
+						domain: domain,
+					},
+					components: {
+						strong: <strong />,
+					},
+				}
+			),
+		};
+	};
+
+	getKeySystemsErrorState = errorMessage => {
+		if (
+			'Invalid attribute value; Contact verification already confirmed, nothing to do' ===
+			errorMessage
+		) {
+			const { domain, email, translate } = this.props;
+
+			return {
+				illustration: '/calypso/images/illustrations/illustration-ok.svg',
+				title: translate( 'Already verified.' ),
+				message: translate(
+					"You've already verified {{strong}}%(email)s{{/strong}} for:{{br /}}{{strong}}%(domain)s{{/strong}}.",
+					{
+						args: {
+							email: email,
+							domain: domain,
+						},
+						components: {
+							strong: <strong />,
+							br: <br />,
+						},
+					}
+				),
+				footer: translate( 'All done. You can close this window now.' ),
+			};
+		}
+	};
+
+	getResendEmailErrorState = () => {
+		const { translate } = this.props;
+
+		return {
+			message: translate( "Sorry, we weren't able to re-send the email." ),
+		};
+	};
+
+	getDefaultErrorState = () => {
+		const { translate } = this.props;
+		const defaultErrorFooter = translate(
+			"If you're having trouble verifying your contact information, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.",
+			{
+				components: {
+					a: <a href={ CALYPSO_CONTACT } />,
+					strong: <strong />,
+				},
+			}
+		);
+
+		return {
+			title: translate( 'Uh oh!' ),
+			illustration: '/calypso/images/illustrations/error.svg',
+			message: translate( 'Hmm. Something went wrong.' ),
+			actionTitle: null,
+			actionCallback: null,
+			footer: defaultErrorFooter,
+			isLoading: false,
+		};
+	};
+
+	setErrorState = error => {
+		let errorState;
+
+		switch ( error.error ) {
+			case 'email_mismatch':
+				errorState = this.getEmailMismatchState();
+				break;
+
+			case 'unauthorized':
+				errorState = this.getExpiredState();
+				break;
+
+			case 'Key_Systems_error':
+				errorState = this.getKeySystemsErrorState( error.message );
+				break;
+
+			case 'resend_email_failed':
+				errorState = this.getResendEmailErrorState();
+				break;
+		}
+
+		this.setState( {
+			...this.getDefaultErrorState(),
+			...errorState,
+		} );
+	};
+
+	resendEmail = () => {
+		const { domain, translate } = this.props;
+
+		this.setState( this.getLoadingState() );
```

Comment:
This state does not really apply here - it shows a `Verifying your contact information…` message, but that's not what's happening and the user could be really confused as a result.

Again, not a blocker, should probably be fixed in a separate PR.